CUDA GPG causes an error

### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

See https://github.com/autowarefoundation/autoware_common/runs/6220775368?check_suite_focus=true#step:6:16

### Expected behavior

No error.

### Actual behavior

```
Run sudo apt-get -yqq update
W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A4B469963BF863CC
E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease' is not signed.
Error: Process completed with exit code 100.
```

### Steps to reproduce

Run the workflow.

### Versions

_No response_

### Possible causes

- Update the key.
- Remove CUDA from the source list after installation.

### Additional context

- https://forums.developer.nvidia.com/t/notice-cuda-linux-repository-key-rotation/212772/1
- https://forums.developer.nvidia.com/t/updating-the-cuda-linux-gpg-repository-key/212897
- https://forums.developer.nvidia.com/t/invalid-public-key-for-cuda-apt-repository/212901
- https://github.com/NVIDIA/nvidia-docker/issues/1631

--- a/ansible/roles/cuda/README.md
+++ b/ansible/roles/cuda/README.md
@@ -20,7 +20,7 @@ For Universe, the `cuda_version` version can also be found in:
 # https://developer.nvidia.com/cuda-11-4-4-download-archive?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=20.04&target_type=deb_network
 wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-ubuntu2004.pin
 sudo mv cuda-ubuntu2004.pin /etc/apt/preferences.d/cuda-repository-pin-600
-sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/7fa2af80.pub
+sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub
 sudo add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/ /"
 sudo apt-get update
 cuda_version=11-4

--- a/ansible/roles/cuda/tasks/main.yaml
+++ b/ansible/roles/cuda/tasks/main.yaml
@@ -17,7 +17,7 @@
 - name: Add NVIDIA apt key
   become: true
   ansible.builtin.apt_key:
-    url: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/{{ cuda_architecture.stdout }}/7fa2af80.pub
+    url: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/{{ cuda_architecture.stdout }}/3bf863cc.pub
 
 - name: Add CUDA repository into sources.list
   become: true

